tests/api/test_result.py

```diff
@@ -16,30 +16,99 @@
 """
 import numpy as np
 import pytest
+from strawberryfields.backends.states import BaseGaussianState
 
 from strawberryfields.result import Result
 
 pytestmark = pytest.mark.api
 
 
+raw_results = {
+    "output": [np.ones((2, 3, 4, 5))],
+    0: [1, 2, 3],
+    1: [4, 5],
+    "other_data": [1, 0, 1, 2],
+    "string": "metadata",
+}
```

Comment:
I know this is just a test module, but there are a few issues with this result:
1. The keys `0` and `1` are illegal since all keys must be strings.
 2. The list values are illegal since all values (with the exception of the `"output"` value) must be NumPy arrays.
 3. The string value `"metadata"` is illegal (see previous point).

Reply:
I'm not sure we ever decided that keys must be strings and values must be arrays, or did we? At least not on the SF side. The idea behind the `samples_dict` was to check for integer keys and have them correspond to modes, and I though metadata could potentially be strings (if needed).

Reply:
Actually, this restriction is enforced by the .npz format which stores data in the form `dict[str, np.ndarray]`.

Reply:
Yeah, real data will never look like this. For example, on QKD_01, raw results presently look like
```
{'alphas': array([-0.00023017-1.17238805e-03j,  0.00163755+1.69466410e-03j,
         0.00170546+1.58876506e-03j, ..., -0.00131527+1.67085418e-03j,
         0.0005287 -7.48599295e-05j,  0.00027381+2.05610324e-04j]),
 'output': [array([ 9.29057598e-04,  1.29526854e-03,  2.39390135e-03, ...,
         -1.32498145e-03,  1.77919865e-05, -1.01980567e-03])]}
```
where "alphas" is the metadata coming from the `.npz` (that we save with the key "alphas").